docs/assets.md

```diff
@@ -0,0 +1,161 @@
+# Asset Management in Whitehall
+
+*Assets, Attachments, Images, Logos, Featured Image, Lead Image, Response Form*
+
+# What is it
+
+There are several features in Whitehall that utilize Images or
+File Attachments.
+
+For example News Articles can have a list of attachments that
+provide further information about the news. Or a Person can have a picture
+which is shown on the person's page and also on any Speech page by that person.
+
+# How did it use to work
+
+Historically Whitehall had all file uploads stored within the Whitehall service. 
+Carrierwave was used to manage the http requests and related file storage within
+Whitehall.
+
+Later a separate service - Asset Manager - became the store for file uploads, taking ownership of every file uploaded.
+Carrierwave was still used to mange the http-request in Whitehall and a plentiful amount of configuration was
+written for carrierwave in order to make AssetManager behave as if it was just a file storage.
+These configurations rippled throughout Whitehall codebase and specialized its behaviour in
+Asset Manager, locking asset manager design and making the overhead of understanding Whitehall
+very heavy. Especially "Legacy url path" was a concept that heavily blocked any refactoring attempts.
+
+"Legacy url path" was Whitehall defining what the url of an asset is, even though Asset Manager would actually
+own the asset. Other publishing applications would let Asset Manager define the url for assets.
+
+Later, the "Legacy url path" was removed and Whitehall Asset references became just simple rows in the
```

Comment:
This is us, right? "At the time of this writing" / "a tech debt team addressed this issue"
There is no immediate connection between the legacy bit and the ADR